cli/src/common.ts

```diff
@@ -324,15 +331,61 @@ export async function printNextSteps(config: Config, appDir: string) {
   log(`Follow the Developer Workflow guide to get building:\n${chalk.bold(`https://capacitor.ionicframework.com/docs/basics/workflow`)}\n`);
 }
 
-export async function checkPlatformVersions(platform: string) {
-  var cliInfo = await runCommand(`npm list @capacitor/cli --json`);
-  var cliVersion = JSON.parse(cliInfo).dependencies["@capacitor/cli"].version;
-  var localInfo = await runCommand(`npm list @capacitor/${platform} --json`);
-  var lovalVersion = JSON.parse(localInfo).dependencies[`@capacitor/${platform}`].version;
+export async function checkPlatformVersions(config: Config, platform: string) {
+  const cliPackagePath = resolveNode(config, '@capacitor/cli', 'package.json');
+  if (!cliPackagePath) {
+    logFatal('Unable to find node_modules/@capacitor/cli/package.json. Are you sure',
+      '@capacitor/cli is installed? This file is currently required for Capacitor to function.');
+    return;
+  }
+
+  const platformPackagePath = resolveNode(config, `@capacitor/${platform}`, 'package.json');
+  if (!platformPackagePath) {
+    logFatal(`Unable to find node_modules/@capacitor/${platform}/package.json. Are you sure`,
+      `@capacitor/${platform} is installed? This file is currently required for Capacitor to function.`);
+    return;
+  }
+
+  const cliVersion = (await readJSON(cliPackagePath)).version;
+  const platformVersion = (await readJSON(platformPackagePath)).version;
 
-  if (semver.gt(cliVersion, lovalVersion)) {
+  if (semver.gt(cliVersion, platformVersion)) {
     log('\n');
     logInfo(`Your @capacitor/cli version is greater than @capacitor/${platform} version`);
     log(`Consider updating to matching version ${chalk`{bold npm install @capacitor/${platform}@${cliVersion}}`}`);
   }
-}
\ No newline at end of file
+}
+
+export function resolveNode(config: Config, ...pathSegments: any[]): string | null {
+  const id = pathSegments[0];
+  const path = pathSegments.slice(1);
+
+  let modulePath;
+  const starts = [config.app.rootDir];
+  for (let start of starts) {
+    modulePath = resolveNodeFrom(start, id);
+    if (modulePath) {
+      break;
+    }
+  }
+  if (!modulePath) {
+    return null;
+  }
+
+  return join(modulePath, ...path);
+}
+
+function resolveNodeFrom(start: string, id: string): string | null {
+  let basePath = resolve(start);
+  let modulePath;
+  while (true) {
+    modulePath = join(basePath, 'node_modules', id);
+    if (existsSync(modulePath)) {
+      return modulePath;
+    }
+    if (basePath === '/') {
```

Comment:
This is was makes the cli hang on windows, as `basePath` will never be `/` on windows.
I've tried getting the base path like this instead of using a hardcoded value
`parse(process.cwd()).root` (needs to add `parse` in the `path` import, and you can remove `relative` from `path` import as it's not used in this file).
I've tried this in windows and mac and seems to work fine

Reply:
Great. I'll do that tonight.

Reply:
Fixed as requested.